### PR TITLE
Remove hideous inline tab indentation, and bans it in contributing guidelines

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
`sed -Ei 's/(\s*\S+)\s*\t+/\1 /g' code/**/*.dm`

image

We have countless examples in the codebase with this style gone wrong, and defines and such being on hideously different levels of indentation. Fixing this to keep the alignment involves tainting the blames of code your PR doesn't need to be touching at all. And ultimately, it's hideous.

There are some files that this sed makes uglier. I can fix these when they are pointed out, but I believe this is ultimately for the greater good of readability. I'm more concerned with if any strings relied on this.

Hi codeowners!

### CODE REVIEW

**Review by Rohesie — APPROVED** [review]
Yes, very much so!

File: code/__DEFINES/achievements.dm [hunk 1]
```diff
@@ -4,102 +4,102 @@
...
 // Medal hub IDs for boss-kill scores
-#define BOSS_SCORE 	         "Bosses Killed"
-#define MINER_SCORE 		 "BDMs Killed"
-#define FROST_MINER_SCORE 	 "DFMs Killed"
-#define BUBBLEGUM_SCORE 	 "Bubblegum Killed"
-#define COLOSSUS_SCORE 	     "Colossus Killed"
-#define DRAKE_SCORE 	     "Drakes Killed"
-#define HIEROPHANT_SCORE 	 "Hierophants Killed"
-#define LEGION_SCORE 	     "Legion Killed"
+#define BOSS_SCORE          "Bosses Killed"
```
Comment (Rohesie) [inline code/__DEFINES/achievements.dm:91]: Guess it didn't catch all cases, chiefly that of multiple spaces.
Can you address that as well on the contribution guidelines change?

**Comment by Rohesie** [comment]
@tgstation/commit-access

File: code/__DEFINES/achievements.dm (hunk 1 above)
Comment (Mothblocks) [reply code/__DEFINES/achievements.dm:91]: Do you not feel the wording of the guidelines is adequate enough? I wrote "tabs/spaces".

Comment (Rohesie) [reply code/__DEFINES/achievements.dm:91]: Nevermind, the wording is fine.
Maybe we should lint for this as well.

**Comment by Jordie0608** [comment]
More like hi code conflicts

**Comment by Mothblocks** [comment]
@Jordie0608 of course, lets get this merged fast then 😄

**Review by spookydonut — APPROVED** [review]

**Review by LemonInTheDark — APPROVED** [review]
Honestly I think indenting defines and such makes them more readable, but you're right about the maintenance cost, and it makes it so much worse when things are mis-indented

**Comment by optimumtact** [comment]
define alignment can be done by sufficiently clever IDE plugins

**Comment by NotRanged** [comment]
> define alignment can be done by sufficiently clever IDE plugins

Know of any for VSC? Indented defines look a lot more readable to me imo

**Review by stylemistake — APPROVED** [review]

**Comment by LemonInTheDark** [comment]
> 
> 
> define alignment can be done by sufficiently clever IDE plugins

Really? Nice

**Pull request merged**